addons/panku_console/modules/variable_tracker/module.gd

```diff
@@ -1,54 +1,128 @@
+	_loop_call_back = _tween_loop.set_loops().tween_callback(_check_current_scene).set_delay(tracking_delay)
+
+
+## Set current scene root node monitoring interval.
+func change_tracking_delay(delay: float) -> void:
+	if _loop_call_back:
+		_loop_call_back.set_delay(delay)
+
+
+# Update current scene root node environment.
+func _check_current_scene() -> void:
+	var scene_root_found: Node = get_scene_root()
+
+	if scene_root_found:
+		if scene_root_found != _current_scene_root:
+			core.gd_exprenv.register_env(CURRENT_SCENE_ENV, scene_root_found)
+			_print_to_interactive_shell(CURRENT_REGISTERED_TIP % [scene_root_found.name, CURRENT_SCENE_ENV])
 
-func get_scene_root() -> Node:
-	var r := core.get_tree().root
-	if r.get_child_count() > _current_scene_index:
-		return r.get_child(_current_scene_index)
 	else:
-		return null
-
-func setup_autoload_tracker():
-	# read root children, the last child is considered as scene node while others are autoloads.
-	var root:Node = core.get_tree().root
-	for i in range(root.get_child_count() - 1):
-		if root.get_child(i).name == core.SingletonName:
-			# skip the plugin singleton
+		if _current_scene_root:
+			core.gd_exprenv.remove_env(CURRENT_SCENE_ENV)
+			_print_to_interactive_shell(CURRENT_REMOVED_TIP % CURRENT_SCENE_ENV)
+
+	_current_scene_root = scene_root_found
+
+## Find the root node of current active scene.
+func get_scene_root() -> Node:
+	# Assuming current scene is the first node in tree that is not autoload singleton.
+	for node:Node in core.get_tree().root.get_children():
+		if not _is_singleton(node):
+			return node
+
+	return null
+
+
+# Find all autoload singletons and bind its to environment vars.
+func _check_autoloads() -> void:
+	var _user_singleton_names := []
+
+	for node:Node in core.get_tree().root.get_children():
+		if node.name == core.SingletonName:
+			# skip panku plugin itself
 			continue
-		# register user singletons
-		var user_singleton:Node = root.get_child(i)
-		core.gd_exprenv.register_env(user_singleton.name, user_singleton)
-
-func print_to_interactive_shell_window():
-	# print a tip to interacvite_shell module
-	# modules load order matters
-	var tip:String = "\n[tip] you can always access current scene by [b]current[/b]"
-	if core.module_manager.has_module("interactive_shell"):
-		var ishell = core.module_manager.get_module("interactive_shell")
-		ishell.interactive_shell.output(tip)
+
+		if _is_singleton(node):
+			# register user singleton
+			_user_singleton_names.append(node.name)
+			core.gd_exprenv.register_env(node.name, node)
+
+	if not _user_singleton_names.is_empty():
+		_print_to_interactive_shell(USER_AUTOLOADS_TIP % ",".join(_user_singleton_names))
+
+
+# Print a tip to interactive shell module, modules load order does matter.
+func _print_to_interactive_shell(message: String) -> void:
+	if core.module_manager.has_module(SHELL_MODULE_NAME):
+		var ishell = core.module_manager.get_module(SHELL_MODULE_NAME)
+		ishell.interactive_shell.output(message)
+
```

Comment:
print something to dev console should be improved later in 1.9.0 to avoid unnecessary setup.